server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -572,7 +583,7 @@ private static IndexStorePlugin.DirectoryFactory getDirectoryFactory(
             throw new IllegalArgumentException("store type [" + storeType + "] is not allowed because mmap is disabled");
         }
         final IndexStorePlugin.DirectoryFactory factory;
-        if (storeType.isEmpty() || isBuiltinType(storeType)) {
```

Comment:
Looks like we've removed the invocation of `isBuiltinType` from most places. Could we just remove the method entirely by removing the remaining invocation? 

https://github.com/opensearch-project/OpenSearch/blob/cd961f39bf57ae92b4486451ce2841b9682c2582/server/src/main/java/org/opensearch/index/IndexModule.java#L565